ota-plus-web/app/reactapp-new/src/store/campaigns/reducers.ts

```diff
@@ -0,0 +1,22 @@
+import {
+  ICampaignsState,
+  ActionType,
+} from './types';
+import { Actions } from './actions';
+
+export const initialState: ICampaignsState = {
+  total: 0,
+  totalWithError: 0
+};
+
+export const campaignsReducer = (state = initialState, action: Actions): ICampaignsState => {
+  switch (action.type) {
+    case ActionType.SET_ALL_CAMPAIGNS_DONE: {
+      return {
+        ...state, ...action.payload
```

Comment:
saga returns { total, totalWithError } so we can just return { ...action.payload } if that's true for every case

Reply:
Done